### PR TITLE
Add support for audit trail

### PR DESCRIPTION
**Pull request by nicola88**
We are adding support for audit logs out of the box in projects based on this template and the incoming version of [lc39](https://github.com/mia-platform/lc39) (v7.2.0).

The PR specifically adds the following:

- a custom log level for generating audit logs, with `audit` as namespace and `1100` as numeric value (the logging stack expects audit logs to have numeric value strictly greater than 1000)
- a custom `logHandler` Pino hook to format and enrich the object passed as first argument to the audit logging method.

### CODE REVIEW

File: docs/30_Audit_Trail.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+Audit logs may contain sensitive information that are not encrypted until they are stored into Mongo, so you must ensure that:
+
+1. Data in transit is always encrypted, by using secure connections between [Kafka][kafka], [Kafka2Rest][kafka2rest], the [CRUD Service][crud-service] and [MongoDB][mongodb]
+2. Only these service can access the audit trail, by using dedicated credentials with exclusive access to the [Kafka][kafka] topic, [MongoDB][mongodb] collection, etc.
+3. Audit trail cannot be altered, deleted or tampered with after being stored into [MongoDB][mongodb]
+
+The following sections illustrate the technical measures you should implement to satisfy these requirements. 
+
+### Encryption of data in transit
```
Comment (calairoh) [inline docs/30_Audit_Trail.md:82]: I guess we're missing a part dedicated to encryption on data in transit on Kafka. Am I wrong? Could we say something on this topic?

File: docs/30_Audit_Trail.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+:::note
+
+Client-side encryption requires an enterprise version of MongoDB supporting [Client-side field level encryption (CSFLE)][mongodb-csfle] (v4.2 or later).
+
+:::
+
+## Configuration
+
+To setup the architecture described previously, you need to configure the following components:
```
Comment (calairoh) [inline docs/30_Audit_Trail.md:107]: I guess we're assuming we always go on the PaaS which already contains Fluentd and Fluentbit. If we're on a customer cluster should we include also configuration of Fluentd or Fluentbit? Or am I loosing something?

Comment (nicola88) [reply docs/30_Audit_Trail.md:107]: I added a short section about the logging operator.

File: docs/30_Audit_Trail.md (hunk 1 above)
Comment (nicola88) [reply docs/30_Audit_Trail.md:82]: Added few more details about Kafka to both this section and the _Encryption of data at rest_ section.

**Review by calairoh — APPROVED** [review]
Comments solved.

**Pull request merged**